src/particles/elements/TaperedPL.H

```diff
@@ -0,0 +1,142 @@
+         *        OR integrated azimuthal magnetic field gradient in T (k > 0)
+         * @param unit  Unit specification
+         *           unit = 0 integrated linear focusing strength in m^(-1)
+         *           unit = 1 integrated linear focusing strength in T
+         * @param taper  horizontal taper parameter in m^(-1)
+                    = 1 / (horizontal dispersion in m)
+         * @param dx horizontal translation error in m
+         * @param dy vertical translation error in m
+         * @param rotation_degree rotation error in the transverse plane [degrees]
+         */
+        TaperedPL (
+            amrex::ParticleReal k,
+            amrex::ParticleReal taper,
+            int unit,
+            amrex::ParticleReal dx = 0,
+            amrex::ParticleReal dy = 0,
+            amrex::ParticleReal rotation_degree = 0
+        )
+        : Alignment(dx, dy, rotation_degree),
+          m_k(k), m_taper(taper), m_unit(unit)
+        {
+        }
+
+        /** Push all particles */
+        using BeamOptic::operator();
+
+        /** This is a tapered PL functor, so that a variable of this type can be used like a
+         *  shortrf function.
+         *
+         * @param x particle position in x
+         * @param y particle position in y
+         * @param t particle position in t
+         * @param px particle momentum in x
+         * @param py particle momentum in y
+         * @param pt particle momentum in t
+         * @param idcpu particle global index (unused)
+         * @param refpart reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void operator() (
+            amrex::ParticleReal & AMREX_RESTRICT x,
+            amrex::ParticleReal & AMREX_RESTRICT y,
+            amrex::ParticleReal & AMREX_RESTRICT t,
+            amrex::ParticleReal & AMREX_RESTRICT px,
+            amrex::ParticleReal & AMREX_RESTRICT py,
+            amrex::ParticleReal & AMREX_RESTRICT pt,
+            [[maybe_unused]] uint64_t & AMREX_RESTRICT idcpu,
+            RefPart const & refpart
+        ) const
+        {
+
+            using namespace amrex::literals; // for _rt and _prt
+
+            // shift due to alignment errors of the element
+            shift_in(x, y, px, py);
+
+            // normalize focusing strength units to MAD-X convention if needed
+            amrex::ParticleReal g = m_k;
+            if (m_unit == 1) {
+                  g = m_k / refpart.rigidity_Tm();
+            }
+
+            // intialize output values
+            amrex::ParticleReal xout = x;
+            amrex::ParticleReal yout = y;
+            amrex::ParticleReal tout = t;
+            amrex::ParticleReal pxout = px;
+            amrex::ParticleReal pyout = py;
+            amrex::ParticleReal ptout = pt;
+
+            // advance position and momentum
+            xout = x;
+            pxout = px - g * ( x + m_taper/2.0 * (pow(x,2) + pow(y,2)) );
```

Comment:
We have to do a general sweep over the code to use `std::pow` (and sin and cos) to avoid promoting always to double:
- C: `pow`/`powf`/... https://en.cppreference.com/w/c/numeric/math/pow
- C++: `std::pow` (with overloads for double/float/...) https://en.cppreference.com/w/cpp/numeric/math/pow
```suggestion
            pxout = px - g * ( x + m_taper*0.5_prt * (std::pow(x, 2) + std::pow(y, 2)) );
```